internal/driver/nodeserver.go

```diff
@@ -29,22 +30,46 @@ import (
 	"k8s.io/mount-utils"
 )
 
-type LinodeNodeServer struct {
-	Driver        *LinodeDriver
-	Mounter       *mount.SafeFormatAndMount
-	DeviceUtils   mountmanager.DeviceUtils
-	CloudProvider linodeclient.LinodeClient
-	Metadata      Metadata
-	Encrypt       Encryption
+type NodeServer struct {
+	driver        *LinodeDriver
+	mounter       *mount.SafeFormatAndMount
+	deviceutils   mountmanager.DeviceUtils
+	client        linodeclient.LinodeClient
+	metadata      Metadata
+	encrypt       Encryption
 	// TODO: Only lock mutually exclusive calls and make locking more fine grained
 	mux sync.Mutex
 
 	csi.UnimplementedNodeServer
 }
 
-var _ csi.NodeServer = &LinodeNodeServer{}
+var _ csi.NodeServer = &NodeServer{}
 
-func (ns *LinodeNodeServer) NodePublishVolume(ctx context.Context, req *csi.NodePublishVolumeRequest) (*csi.NodePublishVolumeResponse, error) {
+func NewNodeServer(linodeDriver *LinodeDriver, mounter *mount.SafeFormatAndMount, deviceUtils mountmanager.DeviceUtils, client linodeclient.LinodeClient, metadata Metadata, encrypt Encryption) (*NodeServer, error) {
+	if linodeDriver == nil {
+		return nil, fmt.Errorf("linodeDriver is nil")
+	}
+	if mounter == nil {
+		return nil, fmt.Errorf("mounter is nil")
+	}
+	if deviceUtils == nil {
+		return nil, fmt.Errorf("deviceUtils is nil")
+	}
+	if client == nil {
+		return nil, fmt.Errorf("linode client is nil")
+	}
```

Comment:
Can we add tests for these checks and verify that they fail with expected errors